src/functions/updateDmsTaskWithDates/framework/dms/dms.ts

```diff
@@ -0,0 +1,276 @@
+/**
+ * Wraps the AWS DMS API.
+ */
+import * as DMS from 'aws-sdk/clients/dms'; // just the DMS apis, not the whole SDK
+import * as escapeJSON from 'escape-json-node';
+import { generateTableMapping, Options } from './table-mapping';
+import { getLogger } from '../util';
+import { config } from '../config/config';
+import { getDmsOptions } from '../config/options';
+import { taskSettings } from '../config/task-settings';
+type UpdateTableMappingCallback = (options: Options) => void;
+
+export class DmsApi {
+  private dms: DMS;
+  private logger = getLogger('DmsApi', 'debug');
+
+  constructor(readonly region: string) {
+    this.dms = new DMS({ apiVersion: '2016-01-01', region: `${region}` });
+  }
+
+  getEndpointArn(identifier: string): Promise<string> {
+    return new Promise<string>((resolve, reject) => {
+
+      const params = {
+        Filters: [{
+          Name: 'endpoint-id',
+          Values: [identifier],
+        }],
+      };
+
+      this.dms.describeEndpoints(params, (err, data) => {
```

Comment:
We should be utilising `async/await` wherever we use the aws-sdk methods wherever we can, i.e.
```
try {
  const data = await this.dms.describeEndpoints(params).promise();
} catch (err} {
  // act on error
}```
Also

Reply:
converted all sdk calls to use promises